include/revng/Pipes/FileContainer.h

```diff
@@ -116,32 +109,41 @@ class FileContainer
     // looking inside the stored file, instead of only checking if the file
     // exists.
     if (this->Path.empty()) {
-      if (llvm::sys::fs::exists(Path))
-        llvm::sys::fs::remove(Path);
+      auto MaybeExists = Path.exists();
+      if (not MaybeExists)
+        return MaybeExists.takeError();
+
+      if (MaybeExists.get())
+        return Path.remove();
 
       return llvm::Error::success();
     }
 
-    if (Path == "-") {
-      auto Buffer = unfailableGetFileAsStream(this->Path);
+    auto MaybeClient = revng::StorageClient::fromURL(this->Path);
+    if (!MaybeClient)
+      return MaybeClient.takeError();
 
-      llvm::outs() << Buffer->getBuffer();
-    }
-
-    auto Error = errorCodeToError(llvm::sys::fs::copy_file(this->Path, Path));
-    auto MaybeError = llvm::sys::fs::getPermissions(this->Path);
-    auto Perm = llvm::cantFail(llvm::errorOrToExpected(std::move(MaybeError)));
-    llvm::sys::fs::setPermissions(Path, Perm);
-    return Error;
+    revng::Path SourcePath(MaybeClient->get());
+    return SourcePath.copyTo(Path);
   }
 
-  llvm::Error loadFromDisk(llvm::StringRef Path) override {
-    if (not llvm::sys::fs::exists(Path)) {
-      *this = FileContainer(this->name());
+  llvm::Error loadFromDisk(const revng::Path &Path) override {
+    auto MaybeExists = Path.exists();
+    if (not MaybeExists)
+      return MaybeExists.takeError();
+
+    if (not MaybeExists.get()) {
+      clear();
       return llvm::Error::success();
     }
+
     getOrCreatePath();
-    return llvm::errorCodeToError(llvm::sys::fs::copy_file(Path, this->Path));
+    auto MaybeClient = revng::StorageClient::fromURL(this->Path);
+    if (!MaybeClient)
+      return MaybeClient.takeError();
+
+    revng::Path DestinationPath(MaybeClient->get());
```

Comment:
1. I'm confused, you're not using `this->Path` here.
2. While I like the cross-client copy infrastructure, I'm not sure it makes a lot of sense to create a `StorageClient` every time. Maybe it would be simpler to just have a `writeFromLocalFile` or `writeFromStream` method.

Reply:
Fixed by the introduction of `Path::fromLocalStorage`